### PR TITLE
fix migration - fixed #5684

### LINKED ISSUE #5684: Errot during migration
https://forum.glpi-project.org/viewtopic.php?id=168415
https://forum.glpi-project.org/viewtopic.php?id=167990
https://forum.glpi-project.org/viewtopic.php?id=168299

### PR DESCRIPTION
**Pull request by yllen**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | /no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #5684 

No complete https://github.com/glpi-project/glpi/issues/2863 check all dbversion value

### CODE REVIEW

File: inc/config.php [hunk 1]
```diff
@@ -211,16 +211,17 @@
                         $newer = true;
                      }
                   }
-               } else if (strlen($CFG_GLPI['dbversion']) > 40) {
+               } else if (!isset($CFG_GLPI['dbversion']) || strlen($CFG_GLPI['dbversion']) > 40) {
```
Comment (cedric-anne) [inline inc/config.php]: Only the following test should be added. The case where `$CFG_GLPI['dbversion']` is not set is already handled on line 222.
```suggestion
               } else if (isset($CFG_GLPI['dbversion']) && strlen($CFG_GLPI['dbversion']) > 40) {
```

Comment (yllen) [reply inc/config.php]: in 222 it's not the same test

214 => test of lenght of dbversion
222 => test of value of dbversion

File: inc/config.php [hunk 2]
```diff
@@ -220,7 +220,7 @@
                   }
                } else if (!isset($CFG_GLPI['dbversion']) || trim($CFG_GLPI["dbversion"]) < GLPI_SCHEMA_VERSION) {
                   $older = true;
-               } else if (trim($CFG_GLPI["dbversion"]) > GLPI_SCHEMA_VERSION) {
+               } else if (!isset($CFG_GLPI['dbversion']) || trim($CFG_GLPI["dbversion"]) > GLPI_SCHEMA_VERSION) {
```
Comment (cedric-anne) [inline inc/config.php]: If `isset($CFG_GLPI['dbversion'])`, the previous `else if` will be true, so this new test is not required.

Comment (yllen) [reply inc/config.php]: Corrected

**Comment by cedric-anne** [comment]
Thanks for the fix.

**Pull request merged**